src/wp-includes/blocks.php

```diff
@@ -707,19 +695,20 @@ function render_block( $parsed_block ) {
 		}
 	}
 
-	/**
-	 * Filters the default context provided to a rendered block.
-	 *
-	 * @since 5.5.0
-	 *
-	 * @param array $context      Default context.
-	 * @param array $parsed_block Block being rendered, filtered by `render_block_data`.
-	 */
-	$context = apply_filters( 'render_block_context', $context, $parsed_block );
-
-	$block = new WP_Block( $parsed_block, $context );
+	$block = new WP_Block(
+		$parsed_block,
+		$context,
+		WP_Block_Type_Registry::get_instance(),
+		array(
+			'is_eager' => false,
+		)
+	);
 
-	return $block->render();
+	return $block->render(
+		array(
+			'filter_data' => true,
+		)
+	);
```

Comment:
Potentially these could be combined into one setting?

```php
$block = new WP_Block(
	$parsed_block,
	$context,
	WP_Block_Type_Registry::get_instance(),
	array(
		'filter_on_render' => true,
	)
);

assert( ! isset( $block->name ) ); // Because `filter_on_render` was provided

$rendered_block = $block->render();

assert( isset( $block->name ) ); // Because `render()` was called
```

Reply:
Alternatively, `filter_data` could be removed. It's probably OK to always call the filters.